vector/src/main/java/im/vector/app/core/resources/DateProvider.kt

```diff
@@ -19,27 +19,30 @@ package im.vector.app.core.resources
 import org.threeten.bp.Instant
 import org.threeten.bp.LocalDateTime
 import org.threeten.bp.ZoneId
+import org.threeten.bp.ZoneOffset
 
 object DateProvider {
 
-    private val zoneId = ZoneId.systemDefault()
-    private val zoneOffset by lazy {
-        val now = currentLocalDateTime()
-        zoneId.rules.getOffset(now)
-    }
+    // recompute the zoneId each time we access it to handle change of timezones
+    private val defaultZoneId: ZoneId
```

Comment:
Might be too costly?

Reply:
I have done this change to handle the case of a change of timezone when the app is running. If we don't recompute the `ZoneId`, in some cases, we don't use the same as the one defined in the system and dates and times are not correct in the app. Do you have another solution in mind to fix this issue?

Reply:
We should probably listen to android.intent.action.TIMEZONE_CHANGED intent if we really want this to be updated. But thats really not a common case, so I don't know?

Reply:
Maybe remove the change waiting for the improvement? What do you think @bmarty?

Reply:
This can be handled later and this can stay like that for now I think.